FluentAvalonia/UI/Controls/ContentDialog/ContentDialog.cs

```diff
@@ -186,9 +186,67 @@ public async Task<ContentDialogResult> ShowAsync(ContentDialogPlacement placemen
                 throw new InvalidOperationException();
 
             ol.Children.Add(_host);
-            (ol.GetVisualRoot() as ILayoutRoot).LayoutManager.ExecuteInitialLayoutPass();            
+            (ol.GetVisualRoot() as ILayoutRoot).LayoutManager.ExecuteInitialLayoutPass();
+        }
+
+        IsVisible = true;
+        ShowCore();
+        SetupDialog();
+        return await _tcs.Task;
+    }
+
+    /// <summary>
+    /// Shows the content dialog on the specified window asynchronously.
+    /// </summary>
+    /// <remarks>
+    /// Note that the placement parameter is not implemented and only accepts <see cref="ContentDialogPlacement.Popup"/>
+    /// </remarks>
+    public async Task<ContentDialogResult> ShowAsync(Window window, ContentDialogPlacement placement = ContentDialogPlacement.Popup)
```

Comment:
To avoid duplicate code:
Make this private and rename it `ShowAsyncCore` 

Then we can have two public methods to call ShowAsyncCore matching the existing and new API (I'm purposefully ignoring the placement arg since it's technically unused, probably should remove it, but that's technically a breaking change):
`public async Task<ContentDialogResult> ShowAsync() => ShowAsyncCore(null)`
and 
`public async Task<ContentDialogResult> ShowAsync(Window w) => ShowAsyncCore(w)`